### PR TITLE
MAUI-860107: UG content for explicit data editors in Data form

### PR DESCRIPTION
**Pull request by Shalini-Ashokan**
**Description**
Add explicit data editors content for data form control.

### CODE REVIEW

File: MAUI/DataForm/explicit-data-editors.md [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+groupItem.Items.Add(new DataFormMultilineItem { FieldName = "Street" });
+groupItem.Items.Add(new DataFormTextItem { FieldName = "State" });
+groupItem.Items.Add(new DataFormTextItem { FieldName = "ZipCode" });
+groupItem.Items.Add(new DataFormAutoCompleteItem { FieldName = "Country", ItemsSource = contactInfoViewModel.CountryNames }); 
+items.Add(groupItem);
+
+dataForm.AutoGenerateItems = false; 
+dataForm.Items = items;
+this.Content = dataForm;
+
```
Comment (karthikraja-arumugam) [inline MAUI/DataForm/explicit-data-editors.md:72]: remove lines

File: MAUI/DataForm/explicit-data-editors.md [hunk 2]
```diff
@@ -0,0 +1,293 @@
+---
+layout: post
+title: Explicit data editors of .NET MAUI DataForm control | Syncfusion
+description: Learn about the explicit data editors of Syncfusion .NET MAUI DataForm(SfDataForm) control in mobile and desktop applications from a single shared codebase.
+platform: Maui
+control: SfDataForm
+documentation: UG
+---
+ 
+# Explicit data editors .NET MAUI DataForm (SfDataForm)
```
Comment (Chrispineagunja) [inline MAUI/DataForm/explicit-data-editors.md:10]: # Explicit Data Editors .NET MAUI DataForm (SfDataForm)

File: MAUI/DataForm/explicit-data-editors.md [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+
+## Explicitly create data editors
+
+The data form auto generates the editors based on the data type, you can explicitly add the data editors by adding [SfDataForm.Items](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_Items) manually and you need to change the [AutoGenerateItems](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_AutoGenerateItems) property to false.
```
Comment (Chrispineagunja) [inline MAUI/DataForm/explicit-data-editors.md:14]: The data form auto-generates the editors based on the data type. You can explicitly add the data editors by adding [SfDataForm.Items](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_Items) manually, and you need to change the [AutoGenerateItems](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_AutoGenerateItems) property to false.

File: MAUI/DataForm/explicit-data-editors.md [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+
+N> Use the [FieldName](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.DataFormItem.html#Syncfusion_Maui_DataForm_DataFormItem_FieldName) property to bind the editor to the data object property.
+[GenerateDataFormItem](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_GenerateDataFormItem) event will not be trigger when [AutoGenerateItems](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_AutoGenerateItems) is false.
```
Comment (Chrispineagunja) [inline MAUI/DataForm/explicit-data-editors.md:17]: The [GenerateDataFormItem](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_GenerateDataFormItem) event will not be triggered when [AutoGenerateItems](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_AutoGenerateItems) is false.

File: MAUI/DataForm/explicit-data-editors.md [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+
+    public string Email { get; set; }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Explicitly add data editor
+
+You can dynamically add the data editor manually by using [Items](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_Items) property of [SfDataForm](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html) when [AutoGenerateItems](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_AutoGenerateItems) is false.
```
Comment (Chrispineagunja) [inline MAUI/DataForm/explicit-data-editors.md:129]: You can dynamically add the data editor manually by using the [Items](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_Items) property of the [SfDataForm](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html) when [AutoGenerateItems](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_AutoGenerateItems) is false.

File: MAUI/DataForm/explicit-data-editors.md [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+{% highlight C# %}
+
+this.dataForm.Items.Add(new DataFormTextItem() { FieldName = "State" });
+this.dataForm.Items.Add(new DataFormTextItem() {FieldName = "Country" });
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Explicitly remove data editor
+You can dynamically remove the data editor explicitly by using the [Items](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_Items) property of [SfDataForm](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html) when [AutoGenerateItems](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_AutoGenerateItems) is false.
```
Comment (Chrispineagunja) [inline MAUI/DataForm/explicit-data-editors.md:141]: You can dynamically remove the data editor explicitly by using the [Items](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_Items) property of the [SfDataForm](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html) when [AutoGenerateItems](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_AutoGenerateItems) is false.

File: MAUI/DataForm/explicit-data-editors.md [hunk 7]
```diff
@@ -0,0 +1,293 @@
...
+{% tabs %}
+{% highlight C# %}
+
+this.dataForm.Items.RemoveAt(2);
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Explicitly clear data editors
+You can dynamically clear all the item views from [SfDataForm](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html) when [AutoGenerateItems](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_AutoGenerateItems) is false.
```
Comment (Chrispineagunja) [inline MAUI/DataForm/explicit-data-editors.md:152]: You can dynamically clear all the item views from the [SfDataForm](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html) when [AutoGenerateItems](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_AutoGenerateItems) is false.

File: MAUI/DataForm/explicit-data-editors.md [hunk 8]
```diff
@@ -0,0 +1,293 @@
...
+{% tabs %}
+{% highlight C# %}
+
+this.dataForm.Items.Clear();
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Explicitly replace data editor
+You can replace data editor with new editor by using the [Items](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_Items) property of [SfDataForm](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html) when AutoGenerateItems is false.
```
Comment (Chrispineagunja) [inline MAUI/DataForm/explicit-data-editors.md:163]: You can replace the data editor with a new editor by using the [Items](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_Items) property of the [SfDataForm](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html) when AutoGenerateItems is false.

File: MAUI/DataForm/explicit-data-editors.md [hunk 9]
```diff
@@ -0,0 +1,293 @@
...
+dataFormGroupItem.Items.Add(new DataFormTextItem() { FieldName = "Last Name" });
+
+this.dataForm.Items.Add(dataFormGroupItem);
+    
+{% endhighlight %}
+{% endtabs %}
+
+#### Explicitly add custom editor
+
+You can add custom editor manually by adding the custom view as an [EditorView](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.DataFormCustomItem.html#Syncfusion_Maui_DataForm_DataFormCustomItem_EditorView) of [DataFormCustomItem](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.DataFormCustomItem.html). In this case, the editor view will not be generated based on field type and [RegisterEditor](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_RegisterEditor_System_String_Syncfusion_Maui_DataForm_DataFormEditorType_).
```
Comment (Chrispineagunja) [inline MAUI/DataForm/explicit-data-editors.md:199]: You can add custom editor manually by adding the custom view as an [EditorView](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.DataFormCustomItem.html#Syncfusion_Maui_DataForm_DataFormCustomItem_EditorView) of [DataFormCustomItem](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.DataFormCustomItem.html). In this case, the editor view will not be generated based on the field type and the [RegisterEditor](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.DataForm.SfDataForm.html#Syncfusion_Maui_DataForm_SfDataForm_RegisterEditor_System_String_Syncfusion_Maui_DataForm_DataFormEditorType_).

File: MAUI/DataForm/getting-started.md [hunk 10]
```diff
@@ -165,6 +165,78 @@ this.Content = dataForm;
 
 N> [View sample in GitHub](https://github.com/SyncfusionExamples/maui-dataform/tree/master/GettingStarted)
 
+## DataForm inside stack layout
+
+#### Vertical StackLayout
+
+When the data form is placed inside a vertical stack layout, users must define the required `MinimumHeightRequest` value. By default, this `MinimumHeightRequest` is set to 300.
```
Comment (Chrispineagunja) [inline MAUI/DataForm/getting-started.md:172]: When the data form is placed inside a vertical stack layout, you must define the required `MinimumHeightRequest` value. By default, this `MinimumHeightRequest` is set to 300.

File: MAUI/DataForm/getting-started.md [hunk 11]
```diff
@@ -165,6 +165,78 @@ this.Content = dataForm;
...
+    MinimumHeightRequest = 400
+};
+this.Content = dataForm;
+
+{% endhighlight %}
+{% endtabs %}
+
+#### Horizontal StackLayout
+
+When the data form is placed inside a horizontal stack layout, users must define the required `MinimumWidthRequest` value. By default, this `MinimumWidthRequest` is set to 300.
```
Comment (Chrispineagunja) [inline MAUI/DataForm/getting-started.md:207]: When the data form is placed inside a horizontal stack layout, you must define the required `MinimumWidthRequest` value. By default, this `MinimumWidthRequest` is set to 300.

**Review by Chrispineagunja — COMMENTED** [review]
Content review completed

File: MAUI/DataForm/explicit-data-editors.md (hunk 1 above)
Comment (Shalini-Ashokan) [reply MAUI/DataForm/explicit-data-editors.md:72]: Removed

File: MAUI/DataForm/explicit-data-editors.md (hunk 2 above)
Comment (Shalini-Ashokan) [reply MAUI/DataForm/explicit-data-editors.md]: Updated

File: MAUI/DataForm/explicit-data-editors.md (hunk 3 above)
Comment (Shalini-Ashokan) [reply MAUI/DataForm/explicit-data-editors.md]: Modified the content

File: MAUI/DataForm/explicit-data-editors.md (hunk 4 above)
Comment (Shalini-Ashokan) [reply MAUI/DataForm/explicit-data-editors.md]: Changed the content

File: MAUI/DataForm/explicit-data-editors.md (hunk 5 above)
Comment (Shalini-Ashokan) [reply MAUI/DataForm/explicit-data-editors.md]: Updated

File: MAUI/DataForm/explicit-data-editors.md (hunk 6 above)
Comment (Shalini-Ashokan) [reply MAUI/DataForm/explicit-data-editors.md]: Modified the content

File: MAUI/DataForm/explicit-data-editors.md (hunk 7 above)
Comment (Shalini-Ashokan) [reply MAUI/DataForm/explicit-data-editors.md]: Changed the content

File: MAUI/DataForm/explicit-data-editors.md (hunk 8 above)
Comment (Shalini-Ashokan) [reply MAUI/DataForm/explicit-data-editors.md]: Updated

File: MAUI/DataForm/explicit-data-editors.md (hunk 9 above)
Comment (Shalini-Ashokan) [reply MAUI/DataForm/explicit-data-editors.md]: Modified the content

File: MAUI/DataForm/getting-started.md (hunk 10 above)
Comment (Shalini-Ashokan) [reply MAUI/DataForm/getting-started.md]: Changed the content

File: MAUI/DataForm/getting-started.md (hunk 11 above)
Comment (Shalini-Ashokan) [reply MAUI/DataForm/getting-started.md]: Updated

**Review by Jeyasri-Murugan — APPROVED** [review]

**Pull request merged**